chia/wallet/trade_record.py

```diff
@@ -47,10 +50,18 @@ def to_json_dict_convenience(self) -> Dict[str, Any]:
         return formatted
 
     @classmethod
-    def from_json_dict_convenience(cls, record: Dict[str, Any], offer: str = "") -> "TradeRecord":
+    def from_json_dict_convenience(
+        cls: Type[_T_TradeRecord], record: Dict[str, Any], offer: str = ""
+    ) -> _T_TradeRecord:
         new_record = record.copy()
         new_record["status"] = TradeStatus[record["status"]].value
         del new_record["summary"]
         del new_record["pending"]
         new_record["offer"] = offer
         return cls.from_json_dict(new_record)
+
+
+@streamable
+@dataclass(frozen=True)
+class TradeRecord(TradeRecordOld):
```

Comment:
Why not just add a new field in the old class?

Reply:
In the DB migration code, I need to be able to de-serialize the old blobs and instead of hand rolling the de-serialization code in that block of code, I figured I would just make a new streamable class and leave the old one available for use during migration purposes.